contracts/bots/PartialLiquidationBot.sol

```diff
@@ -0,0 +1,292 @@
+
+        /// Since we are computing debt and collateral before liquidation,
+        /// we need to update prices beforehand, if needed
+        _applyPriceUpdates(priceUpdates, params.priceOracle);
+
+        _retrieveAndCheckDebt(params);
+
+        _liquidate(params);
+
+        return (params.amountIn, params.amountOut);
+    }
+
+    /// @notice Performs a partial liquidation by swapping some CA collateral to underlying at a discount.
+    ///         Accepts the amount of assetOut that the liquidator wishes to receive
+    /// @param creditManager Credit Manager where the account resides
+    /// @param creditAccount Credit Account to liquidate
+    /// @param assetOut Asset to receive for underlying
+    /// @param amountOut Amount of assetOut to receive. Note that depending on the CA's balance in assetOut and
+    ///                 the maximal swappable amount of underlying, the actual received amount may be less
+    ///                 (with the spent amount adjusted accordingly)
+    /// @param repay Whether to repay debt or only perform a swap
+    /// @param priceUpdates Data for updatable price feeds, if any are required to compute account health
+    function partialLiquidateExactOut(
+        address creditManager,
+        address creditAccount,
+        address assetOut,
+        uint256 amountOut,
+        address to,
+        bool repay,
+        PriceUpdate[] memory priceUpdates
+    ) external returns (uint256, uint256) {
+        LiquidationParams memory params =
+            _prepareParams(creditManager, creditAccount, assetOut, 0, amountOut, to, repay, false);
+
+        /// Since we are computing debt and collateral before liquidation,
+        /// we need to update prices beforehand, if needed
+        _applyPriceUpdates(priceUpdates, params.priceOracle);
+
+        _retrieveAndCheckDebt(params);
+
+        _liquidate(params);
+
+        return (params.amountIn, params.amountOut);
+    }
+
+    /// @dev Internal function that prepares the parameter struct
+    function _prepareParams(
+        address creditManager,
+        address creditAccount,
+        address assetOut,
+        uint256 amountIn,
+        uint256 amountOut,
+        address to,
+        bool repay,
+        bool exactIn
+    ) internal view returns (LiquidationParams memory params) {
+        params.creditManager = creditManager;
+        params.creditAccount = creditAccount;
+        params.assetOut = assetOut;
+        params.amountIn = amountIn;
+        params.amountOut = amountOut;
+        params.exactIn = exactIn;
+        params.repay = repay;
+        params.to = to;
+        params.creditFacade = ICreditManagerV3(params.creditManager).creditFacade();
+        params.cmVersion = ICreditManagerV3(params.creditManager).version();
+        params.underlying = ICreditManagerV3(params.creditManager).underlying();
+        params.priceOracle = ICreditManagerV3(creditManager).priceOracle();
```

Comment:
What the reason to make `priceOracle` call here, if there is no evidence that applyPriceUpdates will be used. Could be called from the function in lazy manner as in CreditFacade

Reply:
It's needed regardless of price updates for exchange rate computation